packages/steps/src/BuildConfig.ts

```diff
@@ -150,12 +150,12 @@ const BuildStepConfigSchema = Joi.any<BuildStepConfig>()
 
 const BuildFunctionConfigSchema = Joi.object({
   name: Joi.string(),
-  platforms: Joi.string().allow(...Object.values(BuildPlatform)),
+  supportedRuntimePlatforms: Joi.array().items(...Object.values(BuildRuntimePlatform)),
   inputs: BuildFunctionInputsSchema,
   outputs: BuildStepOutputsSchema,
   command: Joi.string().required(),
   shell: Joi.string(),
-});
+}).rename('supported_runtime_platforms', 'supportedRuntimePlatforms');
```

Comment:
IIRC we decided on `supported_platforms`. In my opinion, `supported_runtime_platforms` would be too inconvenient to write in a config file. Also, I think it's fine to leave `supportedRuntimePlatforms` in code.
```suggestion
}).rename('supported_platforms', 'supportedRuntimePlatforms');
```